haystack/modeling/data_handler/data_silo.py

```diff
@@ -722,3 +727,68 @@ def get_dict_checksum(payload_dict):
     """
     checksum = hashlib.md5(json.dumps(payload_dict, sort_keys=True).encode("utf-8")).hexdigest()
     return checksum
+
+class DistillationDataSilo(DataSilo):
+    def __init__(self, teacher_model: "FARMReader", teacher_batch_size: int, device, **kwargs):
```

Comment:
I have added the type hint for device. Do you also want me to add type hints for kwargs (i.e. write them out)?

Reply:
Yeah, I think it would be helpful to write them out as it will enable autocomplete in the IDE which can be helpful when users initialize a `DistillationDataSilo` but are not sure which params are expected.